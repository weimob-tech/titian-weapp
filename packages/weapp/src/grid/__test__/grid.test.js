const path = require('path');
const simulate = require('miniprogram-simulate');

describe('test component', () => {
  const id = simulate.load(path.resolve(__dirname, './index'), {
    rootPath: path.join(process.cwd(), 'packages/weapp', 'src')
  });

  it('should render success', async () => {
    const clickCallback = jest.fn();
    const container = simulate.render(id);
    container.addEventListener('itemClick', clickCallback);

    container.attach(document.createElement('parent-wrapper'));
    await simulate.sleep(0);
    expect(container.toJSON()).toMatchSnapshot();

    const item = container.querySelector('.item').querySelector('.titian-grid-item');
    item.dispatchEvent('tap');
    await simulate.sleep(0);
    expect(clickCallback).toBeCalled();
  });
});
