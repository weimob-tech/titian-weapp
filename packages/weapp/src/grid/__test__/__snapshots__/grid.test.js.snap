// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component should render success 1`] = `
<main>
  <ti-grid>
    <wx-view>
      <wx-view
        class="titian-grid-title"
      >
        基本用法
      </wx-view>
      <wx-view
        class="ext-class titian-grid-gutter titian-grid"
        style="padding-left:4px"
      >
        <ti-grid-item
          class="item"
          bind:click="itemClick"
        >
          <wx-view
            class="ext-class titian-grid-item"
            style="width:25%;margin-top:4px;padding-right:4px"
            bind:tap="onClick"
          >
            <wx-view
              class="content-class titian-grid-item-content-gutter titian-grid-item-content"
              style="flex-direction: column"
              bind:tap="onClick"
            >
              <wx-view>
                <ti-icon
                  extClass="icon-class"
                >
                  <wx-view
                    class="titian-icon-base titian-icon titian-icon-home ext-class"
                    style="font-size:56rpx"
                    bind:tap="onClick"
                  />
                </ti-icon>
              </wx-view>
              <wx-view
                class="text-class titian-grid-item-text"
              >
                Grid
              </wx-view>
            </wx-view>
          </wx-view>
        </ti-grid-item>
        <ti-grid-item>
          <wx-view
            class="ext-class titian-grid-item"
            style="width:25%;margin-top:4px;padding-right:4px"
            bind:tap="onClick"
          >
            <wx-view
              class="content-class titian-grid-item-content-gutter titian-grid-item-content"
              style="flex-direction: column"
              bind:tap="onClick"
            >
              <wx-view>
                <ti-icon
                  extClass="icon-class"
                >
                  <wx-view
                    class="titian-icon-base titian-icon titian-icon-home ext-class"
                    style="font-size:56rpx"
                    bind:tap="onClick"
                  />
                </ti-icon>
              </wx-view>
              <wx-view
                class="text-class titian-grid-item-text"
              >
                Grid
              </wx-view>
            </wx-view>
          </wx-view>
        </ti-grid-item>
        <ti-grid-item>
          <wx-view
            class="ext-class titian-grid-item"
            style="width:25%;margin-top:4px;padding-right:4px"
            bind:tap="onClick"
          >
            <wx-view
              class="content-class titian-grid-item-content-gutter titian-grid-item-content"
              style="flex-direction: column"
              bind:tap="onClick"
            >
              <wx-view>
                <ti-icon
                  extClass="icon-class"
                >
                  <wx-view
                    class="titian-icon-base titian-icon titian-icon-home ext-class"
                    style="font-size:56rpx"
                    bind:tap="onClick"
                  />
                </ti-icon>
              </wx-view>
              <wx-view
                class="text-class titian-grid-item-text"
              >
                Grid
              </wx-view>
            </wx-view>
          </wx-view>
        </ti-grid-item>
        <ti-grid-item>
          <wx-view
            class="ext-class titian-grid-item"
            style="width:25%;margin-top:4px;padding-right:4px"
            bind:tap="onClick"
          >
            <wx-view
              class="content-class titian-grid-item-content-gutter titian-grid-item-content"
              style="flex-direction: column"
              bind:tap="onClick"
            >
              <wx-view>
                <ti-icon
                  extClass="icon-class"
                >
                  <wx-view
                    class="titian-icon-base titian-icon titian-icon-home ext-class"
                    style="font-size:56rpx"
                    bind:tap="onClick"
                  />
                </ti-icon>
              </wx-view>
              <wx-view
                class="text-class titian-grid-item-text"
              >
                Grid
              </wx-view>
            </wx-view>
          </wx-view>
        </ti-grid-item>
      </wx-view>
    </wx-view>
  </ti-grid>
  <ti-grid>
    <wx-view>
      <wx-view
        class="titian-grid-title"
      >
        基本用法
      </wx-view>
      <wx-view
        class="ext-class titian-grid-gutter titian-grid"
        style="padding-left:4px"
      />
    </wx-view>
  </ti-grid>
</main>
`;
